src/ArgumentResolver/CategoryByUrlKeyValueResolver.php

```diff
@@ -0,0 +1,86 @@
+<?php
+declare(strict_types = 1);
+namespace App\ArgumentResolver;
+
+use BBC\ProgrammesCachingLibrary\CacheInterface;
+use BBC\ProgrammesPagesService\Domain\Entity\Category;
+use BBC\ProgrammesPagesService\Domain\Entity\Format;
+use BBC\ProgrammesPagesService\Domain\Entity\Genre;
+use BBC\ProgrammesPagesService\Service\CategoriesService;
+use Generator;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Controller\ArgumentValueResolverInterface;
+use Symfony\Component\HttpKernel\ControllerMetadata\ArgumentMetadata;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+/**
+ * If a controller needs a Genre/Format as an argument, resolve it here
+ */
+class CategoryByUrlKeyValueResolver implements ArgumentValueResolverInterface
+{
+    private const SUPPORTED_CLASSES = [
+        // CoreEntity & child classes
+        Genre::class,
+        Format::class,
+        Category::class,
+    ];
+
+    /** @var CategoriesService */
+    private $categoriesService;
+
+    public function __construct(CategoriesService $categoriesService)
+    {
+        $this->categoriesService = $categoriesService;
+    }
+
+    public function supports(Request $request, ArgumentMetadata $argument): bool
+    {
+        return (
+            $request->attributes->has('categoryHierarchy')
+            && $request->attributes->has('categoryType')
+            && in_array($argument->getType(), self::SUPPORTED_CLASSES)
```

Comment:
```suggestion
            && \in_array($argument->getType(), self::SUPPORTED_CLASSES)
```
Looks weird adding a backslash but apparently it helps a bit with opcode optimizations https://github.com/Roave/FunctionFQNReplacer
Not a required changed, mostly a peculiarity I wanted to share about how PHP works